languages/wikipron/src/scrape_and_write.py

```diff
@@ -0,0 +1,123 @@
+import wikipron
+from readme_insert import readme_insert
+from datetime import datetime
+from time import sleep
+import requests
+import os
+import json
+
+
+def call_scrape(lang, config, file_extension):
+    retries = 0
+
+    while retries < 10:
+        file = open(f"../tsv/{lang}{file_extension}.tsv", "w")
+        count = 0
+        try:
+            for (word, pron) in wikipron.scrape(config):
+                count += 1
+                print(f"{word}\t{pron}", file=file)
+            file.close()
+            return count
+        except (
+            requests.exceptions.Timeout,
+            requests.exceptions.ConnectionError
+        ) as err:
+            print(
+                "Timeout or connection error detected during scrape.",
+                lang, file_extension,
+                str(datetime.now()),
+                err
+            )
+            # Pause execution for 30 min.
+            sleep(1800)
+            retries += 1
+            # Need to close file for it to be truncated when re-opening
+            file.close()
+            pass
+
+    file.close()
+    return None
+
+
+def main():
+    languages_file = open("languages.json", "r")
+    LANGUAGES = json.load(languages_file)
+
+    for iso639_code in LANGUAGES:
+        print(
+            'Currently running:',
+            LANGUAGES[iso639_code]["wiktionary_name"],
+            iso639_code,
+            str(datetime.now())
+        )
+
+        row = [
+            iso639_code, LANGUAGES[iso639_code]["iso639_name"],
+            LANGUAGES[iso639_code]["wiktionary_name"],
+            str(LANGUAGES[iso639_code]["casefold"])
+        ]
+
+        phonemic_config = wikipron.Config(
+            key=iso639_code,
+            casefold=LANGUAGES[iso639_code]["casefold"],
+            no_stress=True,
+            no_syllable_boundaries=True
+        )
+        phonetic_config = wikipron.Config(
+            key=iso639_code,
+            casefold=LANGUAGES[iso639_code]["casefold"],
+            phonetic=True,
+            no_stress=True,
+            no_syllable_boundaries=True
+        )
+
+        phonemic_count = call_scrape(iso639_code, phonemic_config, "_phonemic")
+        # Skip phonetic if phonemic failed to complete scrape
+        if phonemic_count is not None:
+            phonetic_count = call_scrape(
+                iso639_code,
+                phonetic_config,
+                "_phonetic"
+            )
```

Comment:
I might be missing something, but why the skip logic? If phonetic exists but phonemic is absent, I'd think we'd still want to scrape for phonetic.

Reply:
What this does is skip calling phonetic if phonemic failed to be scraped within the set amount of retries specified in `call_scrape`. (`call_scrape` returns None in that instance.) 
 
My print messages in these if blocks are a bit confusing because there are two ways we can 'fail to scrape' a language. 
The first, we fail to complete a successful phonemic scrape within the set amount of retries (too many timeout or connection errors), `call_scrape(phonemic settings)` returns None and therefore we don't bother with a phonetic scrape.
The second, we successfully scrape with phonemic and phonetic settings, but the tsvs created are empty and `call_scrape` returns 0 for each (which is the case in your comment below regarding Cantonese).
 
I elected to skip trying a phonetic scrape if we cannot complete a phonemic scrape because it was just simpler to build something that either gets both or nothing.

Reply:
> I elected to skip trying a phonetic scrape if we cannot complete a phonemic scrape because it was just simpler to build something that either gets both or nothing.

I'll address this and other things like adding dialect handling in version 2 while working on the suggestion to separate the scraping and writing aspects of `scrape_and_write.py`.